arm/compute/client.go

```diff
@@ -0,0 +1,2031 @@
+		"skus":           url.QueryEscape(skus),
+		"subscriptionId": url.QueryEscape(client.SubscriptionId),
+		"version":        url.QueryEscape(version),
+	}
+
+	queryParameters := map[string]interface{}{
+		"api-version": ApiVersion,
+	}
+
+	return autorest.DecoratePreparer(
+		client.GetRequestPreparer(),
+		autorest.WithPathParameters(pathParameters),
+		autorest.WithQueryParameters(queryParameters)).Prepare(&http.Request{})
+}
+
+func (client *VirtualMachineImagesClient) GetRequestPreparer() autorest.Preparer {
+	return autorest.CreatePreparer(
+		autorest.AsJSON(),
+		autorest.AsGet(),
+		autorest.WithBaseURL(client.BaseUri),
+		autorest.WithPath("/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers/{publisherName}/artifacttypes/vmimage/offers/{offer}/skus/{skus}/versions/{version}"))
+}
+
+// ListOffers gets a list of virtual machine image offers.;
+//
+func (client *VirtualMachineImagesClient) ListOffers(location string, publisherName string) (result []VirtualMachineImageResource, err error) {
+
+	req, err := client.NewListOffersRequest(location, publisherName)
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure preparing VirtualMachineImagesClient.ListOffers request (%v)", err)
+	}
+
+	req, err = autorest.Prepare(
+		req,
+		client.WithAuthorization(),
+		client.WithInspection())
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure sending VirtualMachineImagesClient.ListOffers request (%v)", err)
+	}
+
+	resp, err := autorest.SendWithSender(client, req)
+	if err != nil {
+		return result, fmt.Errorf("compute: Failure sending VirtualMachineImagesClient.ListOffers request (%v)", err)
+	}
+
+	result = []VirtualMachineImageResource{}
+	err = autorest.Respond(
+		resp,
+		client.ByInspecting(),
+		autorest.WithErrorUnlessOK(),
+		autorest.ByUnmarshallingJSON(&result),
+		autorest.ByClosing())
+
+	return result, err
+}
+
+func (client *VirtualMachineImagesClient) NewListOffersRequest(location string, publisherName string) (*http.Request, error) {
+
+	pathParameters := map[string]interface{}{
+		"location":       url.QueryEscape(location),
+		"publisherName":  url.QueryEscape(publisherName),
+		"subscriptionId": url.QueryEscape(client.SubscriptionId),
+	}
+
+	queryParameters := map[string]interface{}{
+		"api-version": ApiVersion,
+	}
+
+	return autorest.DecoratePreparer(
+		client.ListOffersRequestPreparer(),
+		autorest.WithPathParameters(pathParameters),
+		autorest.WithQueryParameters(queryParameters)).Prepare(&http.Request{})
+}
+
+func (client *VirtualMachineImagesClient) ListOffersRequestPreparer() autorest.Preparer {
+	return autorest.CreatePreparer(
+		autorest.AsJSON(),
+		autorest.AsGet(),
+		autorest.WithBaseURL(client.BaseUri),
+		autorest.WithPath("/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers/{publisherName}/artifacttypes/vmimage/offers"))
+}
+
+// ListPublishers gets a list of virtual machine image publishers.;
+//
```

Comment:
no biggie: Looks like when the method has no parameters to document, the godoc is followed by an extra empty comment like:

```
// MethodName description lorem ipsum.
//
func (....
```